pandas/tests/indexing/test_loc.py

```diff
@@ -2734,6 +2734,17 @@ def test_loc_getitem_nullable_index_with_duplicates():
     tm.assert_series_equal(res, expected)
 
 
+def test_loc_setitem_uint8_upcast():
+    # GH#26049
+
+    df = DataFrame([1, 2, 3, 4], columns=["col1"], dtype="uint8")
+    df.loc[2, "col1"] = 300  # value that can't be held in uint8
+
+    # TODO: would be better to get uint16?
```

Comment:
no i think this is fine, fine-grain upcasting is nice but we don't do it generally

Reply:
The answer is going to end up being "yes" as a side-effect of other upcoming bugfixes.